src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,103 @@
+package duke;
+
+import java.util.Scanner;
+
+/**
+ * Class containing the main Duke application.
+ * Initialises the application.
+ */
+public class Duke {
+    public static void main(String[] args) {
+        Scanner sc = new Scanner(System.in);
+        Storage st = new Storage("./data/dukedata.txt");
+        TaskList tl = new TaskList(st.setFile());
+        Ui ui = new Ui();
+
+        ui.printIntro();
+
+        while (true) {
+            String curr = sc.nextLine();
+            String[] msgWords = curr.split("\\b");
+            for (int i = 0; i < msgWords.length; i++) {
+                if (msgWords[i].equalsIgnoreCase("bye")) {
+                    sc.close();
+                    ui.goodByeMessage();
+                    return;
+                } else if (msgWords[i].equalsIgnoreCase("list")) {
+                    ui.printList(tl);
+                    break;
+                } else if (msgWords[i].equalsIgnoreCase("mark")) {
+                    int index = Integer.parseInt(msgWords[2]) - 1;
+                    if (index < 1 || index > tl.size()) {
+                        ui.printOutOfBoundsMsg();
+                        break;
+                    }
+                    tl.mark(index);
+                    ui.printMarkedMsg(tl.get(index));
+                    break;
+                } else if (msgWords[i].equalsIgnoreCase("unmark")) {
+                    int index = Integer.parseInt(msgWords[2]) - 1;
+                    if (index < 1 || index > tl.size()) {
+                        ui.printOutOfBoundsMsg();
+                        break;
+                    }
+                    tl.unMark(index);
+                    ui.printUnmarkedMsg(tl.get(index));
+                    break;
+                } else if (msgWords[i].equalsIgnoreCase("delete")) {
+                    int index = Integer.parseInt(msgWords[2]) - 1;
+                    Task removed = tl.get(index);
+                    tl.remove(index);
+                    ui.printDeleteMsg(removed.toString(), tl.size());
+                    break;
+                } else if (msgWords[i].equalsIgnoreCase("todo")) {
+                    if (curr.length() < 6) {
```

Comment:
Perhaps using ```length()``` to express these numbers in accordance to the length of the command might be better so that the numbers do not look like "magic numbers".
It might also be better to check if the substring only contains spaces using the ```isBlank()``` method to determine if the task is empty.

This is also found for the subsequent lines of codes with "magic numbers".